xbmcaddon.py

```diff
@@ -225,9 +252,48 @@ def __load_add_on_xml(self):
 
         with io.open(add_on_xml, encoding='utf-8') as fp:
             xml_content = fp.read()
-            self.__version = re.findall(r'version="(\d+.\d+.\d+[^"]*)', xml_content)[0]
+            self.__version = re.findall(r'<addon.*?version="(\d+(?:\.\d)?(?:\.\d)?[^"]*)', xml_content)[0]
             self.__add_on_id = re.findall(r'addon\W+id="([^"]+)"', xml_content)[0]
             self.__name = re.findall(r'name="([^"]+)"', xml_content)[0]
+            tmp = re.findall(r'<addon.*?provider-name="([^"]+)', xml_content)
+            if tmp:
+                self.__author = tmp[0]
+
+            tmp = re.findall(r'<news>(.*?)</news>', xml_content, flags=re.DOTALL)
+            if tmp:
+                self.__news = tmp[0]
+
+            tmp = re.findall(r'<description lang=\"\w+\">(.*?)</description>', xml_content, flags=re.DOTALL)
```

Comment:
I like the language part, so that great. But it not just `description` and `summary` that have this. Also `disclaimer` (see https://kodi.wiki/view/Addon.xml). 

So perhaps we should have a more generic method that extracts the correct language from any of those XML tags? It could take all the match results and return the matched value.

BTW: the xbmc.xbmc.getLanguage() currently always returns "en" or "eng", not "en-GB", that's not what that method does. So I don't think we should use that. Perhaps only match the beginning of the match?

Reply:
You are true, just realized that kodi accept also if no language is supplied.. may do the same also of description and summary? Will check..

Regarding xbmc.getLanguage() I thought that always return eng because is a stub and will be implemeted in the future (detecting from kodi db maybe..), if it's not the case i think it's better to just take the first matching.

Reply:
Even in Kodi it will only return values like "en" or "nl" (ISO_639_1 or ISO_639_2). The addon.xml however requires has language tags like "en-GB" or "nl-NL". So that will never come from the `xbmc.getLanguage()`. So i would just find the first "en-...." and take that one, or if that one is not present, just the first one.